version.py

```diff
@@ -36,37 +36,61 @@
 
 import pathlib
 import subprocess
+import sys
 VERSION_FILE = "RELEASE-VERSION"
 
 
 def call_git_describe(prefix='', cwd='.'):
-    cmd = 'git describe --tags --match %s[0-9]*' % prefix
-    try:
-        version = subprocess.run(
-            cmd.split(),
-            stdout=subprocess.PIPE,
-            check=True,
-        ).stdout.decode()
-        version = version.strip()[len(prefix):]
-        if '-' in version:
-            version = '{}+dev{}.{}'.format(*version.split('-'))
-        return version
-    except Exception:
-        return None
-
-
-def get_version(prefix='', cwd='.'):
+    version_cmd = 'git describe --tags --match %s[0-9]*' % prefix
+    last_tag_cmd = 'git describe --tags --match %s[0-9]* --abbrev=0' % prefix
+    version = subprocess.run(
+        version_cmd.split(),
+        stdout=subprocess.PIPE,
+        check=True,
+        shell=False,
+    ).stdout.decode()
+
+    last_tag = subprocess.run(
+        last_tag_cmd.split(),
+        stdout=subprocess.PIPE,
+        check=True,
+        shell=False,
+    ).stdout.decode()
+
+    assert ' ' not in prefix
+    assert version.strip().startswith(prefix)
+    assert last_tag.strip().startswith(prefix)
+
+    version = version.strip()[len(prefix):]
+    last_tag = last_tag.strip()[len(prefix):]
+
+    assert version.startswith(last_tag)
+    if version != last_tag:
+        # `version` does not match the last tag so we must be on an untagged commit.
+        # In that case `version` is that tag with a suffix consisting of the number of commits and a commit ID.
+        # To make this very clear, we want to insert `+dev` between the tag and the suffix. Also, replace the
+        # hyphen in the suffix with a dot but without mangling any hyphens that might be in the tag.
+        # In effect something like 0.5.3-rc5-post1-5-78df3e12 becomes 0.5.3-rc5-post1+dev5.78df3e12.
+        version = f"{last_tag}+dev{version[len(last_tag) + 1:].replace('-', '.')}"
+    return version
+
+
+def get_version(prefix='', cwd='.', update_version_file=True):
     path = pathlib.Path(cwd) / VERSION_FILE
     try:
         with path.open("r") as f:
             release_version = f.read()
-    except Exception:
+    except FileNotFoundError:
         release_version = None
 
+    if not update_version_file:
+        if not release_version:
```

Comment:
This should be `if release_version is None:`. Or `if release_version in [None, '']:` if you really wanted to raise the error when the file is empty.